server/graphql/invoice/src/mutations/outbound_return/insert.rs

```diff
@@ -34,12 +42,31 @@ pub enum InsertResponse {
     Response(InvoiceNode),
 }
 
-pub fn insert(_ctx: &Context<'_>, _input: InsertInput) -> Result<InsertResponse> {
-    Ok(InsertResponse::Response(InvoiceNode::from_domain(
-        Invoice {
-            ..Default::default()
+pub fn insert(ctx: &Context<'_>, store_id: &str, input: InsertInput) -> Result<InsertResponse> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            // resource: Resource::MutateOutboundReturn, // TODO
+            resource: Resource::MutateInboundShipment,
+            store_id: Some(store_id.to_string()),
         },
-    )))
+    )?;
+
+    let service_provider = ctx.service_provider();
+    let service_context = service_provider.context(store_id.to_string(), user.user_id)?;
+
+    let result = service_provider
+        .invoice_service
+        .insert_outbound_return(&service_context, input.to_domain());
+
+    match result {
```

Comment:
Sometimes verboseness of rust makes it a bit harder to digest the logic, 

I like to return result here and then Ok(result), rather then Ok() in each match. This is stylistic preference though, nothing prescriptive